plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -630,3 +638,25 @@ func (p *Plugin) admitCSINode(nodeName string, a admission.Attributes) error {
 
 	return nil
 }
+
+func (p *Plugin) admitNodeResourceSlice(nodeName string, a admission.Attributes) error {
+	// Shortcut the check if not enabled.
+	if !p.dynamicResourceAllocationEnabled {
+		return nil
```

Comment:
hmm, if the feature isn't enabled, why are nodes writing to this API? shouldn't we be guarding any requests we see to this API, and counting on none showing up if the feature is disabled?

Reply:
They shouldn't. If they did, the apiserver would reject the request because it doesn't serve the resource. This makes this check unnecessary when the feature is disabled.

The rationale for adding the check was that if a node tries to create while the feature is disabled, we don't want to run the code below that is conceptually alpha quality. I don't have a strong preference, though, and can also remove this check.

Reply:
> if a node tries to create while the feature is disabled, we don't want to run the code below that is conceptually alpha quality

if a request gets here, that means it is being served (which would surprise me)

if it is being served AND a node is trying to create it AND we *don't* protect the API, that seems bad.

Reply:
So just remove the if check?

Reply:
Done.